pkg/cmd/workspace/create.go

```diff
@@ -467,14 +467,34 @@ func waitForDial(workspace *apiclient.Workspace, activeProfile *config.Profile,
 		}
 	}
 
-	for {
-		dialConn, err := tsConn.Dial(context.Background(), "tcp", fmt.Sprintf("%s:%d", project.GetProjectHostname(workspace.Id, workspace.Projects[0].Name), ssh_config.SSH_PORT))
-		if err == nil {
-			return dialConn.Close()
+	connectChan := make(chan error)
+	timeout := time.After(10 * time.Second)
+
+	go func() {
+		for {
+			dialConn, err := tsConn.Dial(context.Background(), "tcp", fmt.Sprintf("%s:%d", project.GetProjectHostname(workspace.Id, workspace.Projects[0].Name), ssh_config.SSH_PORT))
+			if err == nil {
+				connectChan <- dialConn.Close()
+				return
+			}
+			time.Sleep(time.Second)
 		}
+	}()
 
-		time.Sleep(time.Second)
+	time.Sleep(3 * time.Second)
+	err := views_util.WithInlineSpinner("Connection to tailscale is taking longer than usual", func() error {
+		select {
+		case <-connectChan:
+			return <-connectChan
```

Comment:
Should be `case err := <- connectChan` and `return err`. Once a value is consumed from the channel, it's gone.

Reply:
At first did it like that, but changed it because this looked cleaner. Will revert. 👍